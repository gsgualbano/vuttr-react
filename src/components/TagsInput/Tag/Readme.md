Tag exemplo:

```js
<Tag>Exemplo</Tag>
```
