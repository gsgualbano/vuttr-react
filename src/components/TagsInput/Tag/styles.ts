import styled from 'styled-components';

export const Container = styled.div`
  display: inline-flex;
  flex-direction: row;
  background: #f95e5a;
  border-radius: 13px;
  padding: 2px;
  margin: 2px;
  color: #ffffff;
`;
