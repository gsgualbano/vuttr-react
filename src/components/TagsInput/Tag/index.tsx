import React from 'react';

import { Container } from './styles';

/**
 * Propriedades da Tag
 */
interface OwnProps {
  /** elemento jsx */
  children: React.ReactNode;
}

/**
 * Tag badge.
 */
const Tag = ({ children }: OwnProps): JSX.Element => <Container>{children}</Container>;

export default Tag;
