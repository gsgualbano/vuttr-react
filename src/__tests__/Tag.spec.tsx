import React from 'react';
import { render } from '@testing-library/react';
import Tag from '~/components/TagsInput/Tag';
import 'jest-dom/extend-expect';
import 'jest-styled-components';

describe('<Tag />', () => {
  it('render correctly', () => {
    const { asFragment } = render(<Tag>Google</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render with text', () => {
    const { getByText } = render(<Tag>Bing</Tag>);
    expect(getByText('Bing')).toBeInTheDocument();
  });
});
