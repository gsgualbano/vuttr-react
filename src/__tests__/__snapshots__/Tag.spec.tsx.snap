// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> render correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #f95e5a;
  border-radius: 13px;
  padding: 2px;
  margin: 2px;
  color: #ffffff;
}

<div
    class="c0"
  >
    Google
  </div>
</DocumentFragment>
`;
